### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by singhpratyush**
After merging this patch, introducing new dependencies will no longer need to add new lines in `.travis.yml`. Also, the scripts introduced can be used for local testing setup.

* Add scripts for installing apt, npm and go dependencies
* Add requirement list for python, apt and npm
* Remove yml bear
* Add .idea to .gitignore
* Test running of each code
* Fail on `.editorconfig` violation.

### CODE REVIEW

File: merge_sort/merge_sort.c [hunk 1]
```diff
@@ -54,7 +54,7 @@ int main() {
 	int arr[] = {7, 10, 14, 1, 2, 6, 17, 0};
 	int size = sizeof(arr) / sizeof(arr[0]);
 	merge_sort(arr, 0, size - 1);
-	for (i = 0; i < n; i++) {
+	for (i = 0; i < size; i++) {
```
Comment (aviaryan) [inline merge_sort/merge_sort.c]: 😧

**Comment by singhpratyush** [comment]
@aviaryan: Ready for review.

File: dijkstra/dijkstra.c [hunk 2]
```diff
@@ -16,7 +16,7 @@
 int minimum_distance(const int min_distances[], const bool shortest_paths[]) {
 	int i;
 	int minimum = INFINITY, index;
-	for (int i = 0; i < VERTICES; ++i) {
+	for (i = 0; i < VERTICES; ++i) {
```
Comment (singhpratyush) [inline dijkstra/dijkstra.c]: @Monal5031: This implementation had a compile time error, yet you submitted it. This means that you didn't find it necessary to compile a code before submitting a patch, which is bad. Please work in it.

File: euclidean_gcd/euclidean_gcd.c [hunk 3]
```diff
@@ -10,7 +10,7 @@ int euclidean_gcd_recursive(int first, int second) {
         if(second == 0)
             return first;                // First becomes gcd if second becomes zero
         else
-            return gcd_recursive(second, (first % second));
+            return euclidean_gcd_recursive(second, (first % second));
```
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: @Monal5031: Same mistake here.

File: merge_sort/merge_sort.c (hunk 1 above)
Comment (singhpratyush) [inline merge_sort/merge_sort.c]: @Monal5031: And here too.

File: sieve_of_eratosthenes/sieve_of_eratosthenes.py [hunk 4]
```diff
@@ -7,20 +7,20 @@ def sieve_of_eratosthenes(n):
     :param n: Number upto which to calculate primes
     :return: A boolean list contaning only primes
     """
-    primes = [True] * (n+1)  # Setting all as True initially
+    primes = [True] * (n + 1)  # Setting all as True initially
     primes[0] = primes[1] = False  # Since 0 and 1 are not primes
-    sqrt_n = math.ceil(math.sqrt(n))
+    sqrt_n = int(math.ceil(math.sqrt(n)))
```
Comment (singhpratyush) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.py]: @Monal5031: Always make sure that you check for version compatibility of Python code.

File: heap_sort/heap_sort.c [hunk 5]
```diff
@@ -8,53 +8,53 @@
...
-    if ((r < heapsize) && (a[r] > a[largest]))
-         largest = r ;
-    // If root is not the largest.
-    if (largest != root) {
-         int tmp = a[root];
-         a[root] = a[largest];
-         a[largest] = tmp;
-         max_heapify(a, largest, heapsize);
-    }
+		int largest = root;
```
Comment (aviaryan) [inline heap_sort/heap_sort.c]: Do we have an indentation of 2 tabs here ?

File: insertion_sort/insertion_sort.c [hunk 6]
```diff
@@ -3,33 +3,33 @@
...
-          of their current position */
-       while (j >= 0 && arr[j] > key) {
-           arr[j+1] = arr[j];
-           j--;
-       }
-       arr[j+1] = key;
-   }
+	 int i, key, j;
+	 for (i = 1; i < arr_size; i++) {
+			 key = arr[i];
```
Comment (aviaryan) [inline insertion_sort/insertion_sort.c]: 2 tabs?

File: prims/prims.c [hunk 7]
```diff
@@ -6,42 +6,42 @@
...
-    int i, j;
-    int parent[SIZE];
-    memset(parent, 0, SIZE);
-    printf("Edge\tWeight\n");
-    prims(G,parent);
-    for (i = 1; i < SIZE; ++i)
-        printf("%d - %d\t%d \n", parent[i], i, G[i][parent[i]]);
-    return 0;
+		int G[SIZE][SIZE] = {{0, 2, 0, 6, 0},
+												 {2, 0, 3, 8, 5},
```
Comment (aviaryan) [inline prims/prims.c]: Why does the diff looks so bad here.

File: dijkstra/dijkstra.c (hunk 2 above)
Comment (Monal5031) [inline dijkstra/dijkstra.c]: @singhpratyush I checked it twice sir, don't know how it is showing it here? I even tested on various inputs. 😕

File: sieve_of_eratosthenes/sieve_of_eratosthenes.py (hunk 4 above)
Comment (Monal5031) [inline sieve_of_eratosthenes/sieve_of_eratosthenes.py]: Okay sir will make sure of this one from next time. Sorry for inconvenience.

File: dijkstra/dijkstra.c (hunk 2 above)
Comment (singhpratyush) [inline dijkstra/dijkstra.c]: I don't understand how we are not supposed to see an error here. If you have already declared `i` in the beginning, how can you declare it in the `for` loop again?

Comment (Monal5031) [inline dijkstra/dijkstra.c]: I have missed that one while typing but it didn't show error locally or even on ideone, I'm also confused about how this happened logically it should show out error but it didn't.

File: euclidean_gcd/euclidean_gcd.c [hunk 8]
```diff
@@ -7,26 +7,27 @@
 */
 
 int euclidean_gcd_recursive(int first, int second) {
-        if(second == 0)
-            return first;                // First becomes gcd if second becomes zero
-        else
-            return gcd_recursive(second, (first % second));
+		if(second == 0)
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.c]: Two tabs?

File: euclidean_gcd/euclidean_gcd.c [hunk 9]
```diff
@@ -7,26 +7,27 @@
...
 }
 
 int euclidean_gcd(int first, int second) {
-        while(second != 0) {            // Iterate till second becomes zero
-            int temp = second;          // Temporary variable to hold value of second which is to be assigned to first later
-            second = first % second;
-            first = temp;
-        }
-        return first;                   // When second becomes 0, first becomes gcd of both
+		while(second != 0) {            // Iterate till second becomes zero
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.c]: Same here.

File: euclidean_gcd/euclidean_gcd.c [hunk 10]
```diff
@@ -7,26 +7,27 @@
...
 }
 
 int main() {
-        int first = 49;
-        int second = 7;
-        int answer_recursive = euclidean_gcd_recursive(first, second);
-        int answer_iterative = euclidean_gcd(first, second);
-        printf("GCD of %d and %d is : %d by recursive algo.\n", first, second, answer_recursive);
-        printf("GCD of %d and %d is : %d by iterative algo.\n", first, second, answer_iterative);
+		int first = 49;
```
Comment (aviaryan) [inline euclidean_gcd/euclidean_gcd.c]: And here.

File: n_queen_problem/NQueenProblem.cpp [hunk 11]
```diff
@@ -18,54 +18,54 @@ using namespace std;
...
+		int i,j;
+		for (i = 0; i < col; i++)
+				if (board[row][i]==1)
+						return false;  // * return false, if there's another Queen present in the same row.
 
-    for (i=row, j=col; i>=0 && j>=0; i--, j--)
-        if (board[i][j]==1)
-            return false;  // * return false, if there's another Queen present in the upper diagonal.
+		for (i=row, j=col; i>=0 && j>=0; i--, j--)
+				if (board[i][j]==1)
```
Comment (aviaryan) [inline n_queen_problem/NQueenProblem.cpp]: Two tabs

File: euclidean_gcd/euclidean_gcd.c (hunk 8 above)
Comment (singhpratyush) [inline euclidean_gcd/euclidean_gcd.c]: I guess this is with all files that are fixed with `editorconfig-tools`.

File: sieve_of_eratosthenes/sieveOfEratosthenes.c [hunk 12]
```diff
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <string.h>
-#include <math.h>     // for using square root 
-#include <stdbool.h>  // to include bool datatype 
+#include <math.h>     // for using square root
+#include <stdbool.h>  // to include bool datatype
 
 void sieveOfEratosthenes(int n, bool *primes) {
 	int i, sqrtOfn = sqrt(n)+1, j;
```
Comment (Monal5031) [inline sieve_of_eratosthenes/sieveOfEratosthenes.c]: Please add space around binary operator here.

Comment (singhpratyush) [inline sieve_of_eratosthenes/sieveOfEratosthenes.c]: We need a proper configurable linter for this.

Comment (Monal5031) [inline sieve_of_eratosthenes/sieveOfEratosthenes.c]: Yeah we can use that, but we'll need to configure it ourself.

**Pull request merged**